src/rust/src/tests/ed25519.rs

```diff
@@ -1,50 +1,735 @@
+        if i % 2 == 0 {
+            l = c.to_digit(16).unwrap();
+        } else {
+            tmp[i / 2] = (l << 4 | c.to_digit(16).unwrap()) as u8;
+        };
+    }
+    tmp
+}
+
+fn hex_sig(s: &str) -> [u8; 64] {
+    let mut tmp = [0u8; 64];
+    let mut l = 0;
+    for (i, c) in s.chars().enumerate() {
+        if i % 2 == 0 {
+            l = c.to_digit(16).unwrap();
+        } else {
+            tmp[i / 2] = (l << 4 | c.to_digit(16).unwrap()) as u8;
+        };
+    }
+    tmp
+}
+
 #[test]
-fn test_weird_signature() {
-    // This is a signature from hell.
+fn test_hex() {
+    assert!(
+        &(hex_vk("c7176a703d4dd84fba3c0b760d10670f2a2053fa2c39ccc64ec7fd7792ac037a"))[..]
+            == &[
+                0xc7, 0x17, 0x6a, 0x70, 0x3d, 0x4d, 0xd8, 0x4f, 0xba, 0x3c, 0x0b, 0x76, 0x0d, 0x10,
+                0x67, 0x0f, 0x2a, 0x20, 0x53, 0xfa, 0x2c, 0x39, 0xcc, 0xc6, 0x4e, 0xc7, 0xfd, 0x77,
+                0x92, 0xac, 0x03, 0x7a,
+            ]
+    );
+    assert!(
+        &(hex_sig("c7176a703d4dd84fba3c0b760d10670f2a2053fa2c39ccc64ec7fd7792ac037ac7176a703d4dd84fba3c0b760d10670f2a2053fa2c39fff64ec7fd7792ac047a"))[..] ==
+        &([
+            0xc7, 0x17, 0x6a, 0x70, 0x3d, 0x4d, 0xd8, 0x4f, 0xba, 0x3c, 0x0b, 0x76, 0x0d, 0x10, 0x67, 0x0f, 0x2a, 0x20, 0x53, 0xfa, 0x2c, 0x39, 0xcc, 0xc6, 0x4e, 0xc7, 0xfd, 0x77, 0x92, 0xac, 0x03, 0x7a,
+            0xc7, 0x17, 0x6a, 0x70, 0x3d, 0x4d, 0xd8, 0x4f, 0xba, 0x3c, 0x0b, 0x76, 0x0d, 0x10, 0x67, 0x0f, 0x2a, 0x20, 0x53, 0xfa, 0x2c, 0x39, 0xff, 0xf6, 0x4e, 0xc7, 0xfd, 0x77, 0x92, 0xac, 0x04, 0x7a,
+        ])[..]
+    );
+}
 
-    // A = (0, 1) encoded with high bit set, even though the x-coordinate is 0
-    let pk = [
-        0x01, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x80
-    ];
+#[test]
+fn test_vectors() {
+    fn run_test_vector(vk: &str, sig: &str) {
+        let vk = hex_vk(vk);
+        let sig = hex_sig(sig);
 
-    // R = (0, -1) encoded with high bit set, even though the x-coordinate is 0
-    // s = 0
-    let mut sig = [
-        0xec, 0xff, 0xff, 0xff,
-        0xff, 0xff, 0xff, 0xff,
-        0xff, 0xff, 0xff, 0xff,
-        0xff, 0xff, 0xff, 0xff,
-        0xff, 0xff, 0xff, 0xff,
-        0xff, 0xff, 0xff, 0xff,
-        0xff, 0xff, 0xff, 0xff,
-        0xff, 0xff, 0xff, 0xff,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00,
-        0x00, 0x00, 0x00, 0x00
-    ];
-    
-    assert_eq!(librustzcash_zebra_crypto_sign_verify_detached(
-        &sig, &0xff, 1, &pk
-    ), 0);
+        assert!(librustzcash_zebra_crypto_sign_verify_detached(&sig, &0xff, 0, &vk) == 0);
```

Comment:
How easy would it be to also verify the test vectors against the current libsodium behavior?

Reply:
Not hard, it would just involve modifying [these existing tests.](https://github.com/zcash/zcash/blob/master/src/gtest/test_consensus.cpp) I can toss that into this PR, since I'm blocked on review anyway.

Reply:
I just moved the test vectors there in the latest commit.